org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ChangeConverter.java

```diff
@@ -0,0 +1,245 @@
+			}
+			checkDerived(file);
+			List<ReplaceEdit> textEdits = change.getReplacements().stream().map(replacement -> {
+				return new ReplaceEdit(replacement.getOffset(), replacement.getLength(), replacement.getReplacementText());
+			}).collect(Collectors.toList());
+
+			MultiTextEdit textEdit = new MultiTextEdit();
+			textEdit.addChildren(textEdits.toArray(new TextEdit[textEdits.size()]));
+			ITextEditor openEditor = findOpenEditor(file);
+			final TextChange ltkChange;
+			if (openEditor == null) {
+				TextFileChange textFileChange = new TextFileChange(change.getOldURI().lastSegment(), file);
+				textFileChange.setSaveMode(TextFileChange.FORCE_SAVE);
+				ltkChange = textFileChange;
+			} else {
+				ltkChange = new EditorDocumentChange(currentChange.getName(), openEditor, false);
+			}
+			ltkChange.setEdit(textEdit);
+			ltkChange.setTextType(change.getOldURI().fileExtension());
+			addChange(ltkChange);
+		}
+	}
+
+	protected void handleUriChange(IEmfResourceChange change) {
+		if (!change.getNewURI().equals(change.getOldURI())) {
+			if (change.getNewURI().lastSegment().equals(change.getOldURI().lastSegment())) {
+				IFile oldFile = resourceUriConverter.toFile(change.getOldURI());
+				if (!canWrite(oldFile)) {
+					issues.add(RefactoringIssueAcceptor.Severity.FATAL, "Cannot move read-only file '" + oldFile.getFullPath() + "'");
+				}
+				checkDerived(oldFile);
+				IFile newFile = resourceUriConverter.toFile(change.getNewURI());
+				IContainer newContainer = newFile.getParent();
+				MoveResourceChange ltkChange = new MoveResourceChange(oldFile, newContainer);
+				addChange(ltkChange);
+			} else {
+				if (change.getNewURI().trimSegments(1).equals(change.getOldURI().trimSegments(1))) {
+					RenameResourceChange ltkChange = new RenameResourceChange(
+							resourceUriConverter.toFile(change.getOldURI()).getFullPath(), change.getNewURI().lastSegment());
+					addChange(ltkChange);
+				}
+			}
+		}
+	}
+
+	protected void addChange(Change change) {
+		if (changeFilter == null || changeFilter.apply(change)) {
+			currentChange.add(change);
+		}
+	}
+
+	protected boolean canWrite(IFile file) {
+		return file.getWorkspace().validateEdit(new IFile[] { file }, IWorkspace.VALIDATE_PROMPT).isOK();
+	}
+
+	protected void checkDerived(IFile file) {
+		if (file.isDerived()) {
+			issues.add(RefactoringIssueAcceptor.Severity.WARNING, "Affected file '" + file.getFullPath() + "' is derived");
+		}
+	}
+
+	protected ITextEditor findOpenEditor(IFile file) {
+		if (workbench == null) {
+			return null;
+		}
+
+		return new DisplayRunnableWithResult<ITextEditor>() {
+
+			@Override
+			protected ITextEditor run() throws Exception {
+				FileEditorInput editorInput = new FileEditorInput(file);
+				IEditorReference[] editorReferences = workbench.getActiveWorkbenchWindow().getActivePage().getEditorReferences();
+				Optional<ITextEditor> textEditorPart = Arrays.asList(editorReferences) //
```

Comment:
A general remark: Too many streams for my liking.
Eclipse stream debugging support is not ready for prime time yet and thus I tend to rewrite these to normal for loops more often than I enjoy having the stream API

Reply:
@cdietrich Wasn't there a util to find an open editor for an editor input?

Reply:
@szarnekow `workbench.getActiveWorkbenchWindow().getActivePage().findEditor` can be used here. The logic can be simplified to